# HG changeset patch
# User maorshutman <maorus12@gmail.com>
# Date 1521713034 -7200
#      Thu Mar 22 12:03:54 2018 +0200
# Branch stable
# Node ID ef8df0a8b358f8acf575811c07ad39bd1653de03
# Parent  90f3716866717aceda7e51de60c673bc91f8b7aa
Add element-by-element multiplication operator for ranges (bug #53232).

* op-range.cc: Use INSTALL_BINOP_TI to map element-by-element multiplication
operator to ordinary multiplication.

diff --git a/libinterp/operators/op-range.cc b/libinterp/operators/op-range.cc
--- a/libinterp/operators/op-range.cc
+++ b/libinterp/operators/op-range.cc
@@ -103,17 +103,19 @@ install_range_ops (octave::type_info& ti
   INSTALL_UNOP_TI (ti, op_hermitian, octave_range, transpose);
 
   INSTALL_BINOP_TI (ti, op_add, octave_range, octave_scalar, addrs);
   INSTALL_BINOP_TI (ti, op_add, octave_scalar, octave_range, addsr);
   INSTALL_BINOP_TI (ti, op_sub, octave_range, octave_scalar, subrs);
   INSTALL_BINOP_TI (ti, op_sub, octave_scalar, octave_range, subsr);
   INSTALL_BINOP_TI (ti, op_mul, octave_range, octave_scalar, mulrs);
   INSTALL_BINOP_TI (ti, op_mul, octave_scalar, octave_range, mulsr);
-
+  
+  INSTALL_BINOP_TI (ti, op_el_mul, octave_range, octave_scalar, mulrs);
+  INSTALL_BINOP_TI (ti, op_el_mul, octave_scalar, octave_range, mulsr);
   INSTALL_BINOP_TI (ti, op_el_pow, octave_scalar, octave_range, el_powsr);
   INSTALL_BINOP_TI (ti, op_el_pow, octave_complex, octave_range, el_powcsr);
 
   INSTALL_CATOP_TI (ti, octave_range, octave_range, r_r);
   INSTALL_CATOP_TI (ti, octave_range, octave_scalar, r_s);
   INSTALL_CATOP_TI (ti, octave_range, octave_matrix, r_m);
   INSTALL_CATOP_TI (ti, octave_range, octave_complex, r_cs);
   INSTALL_CATOP_TI (ti, octave_range, octave_complex_matrix, r_cm);
